packages/components/src/ClipboardButton/ClipboardButton.js

```diff
@@ -0,0 +1,45 @@
+import { contextConnect, useContextSystem } from '@wp-g2/context';
+import { useClipboard } from '@wp-g2/utils';
+import { noop } from 'lodash';
+import React from 'react';
+import { useEffect } from 'react';
+import { useRef } from 'react';
+
+import { Button } from '../Button';
+
+function ClipboardButton(props, forwardedRef) {
+	const {
+		text,
+		onClick: onClickProp = noop,
+		onCopy: onCopyProp = noop,
+		onFinishCopy = noop,
+		...otherProps
+	} = useContextSystem(props, 'ClipboardButton');
+	const { hasCopied, onCopy } = useClipboard(text);
+	const lastHasCopied = useRef(hasCopied);
+
+	useEffect(() => {
+		if (lastHasCopied.current === hasCopied) {
+			return;
+		}
+
+		if (hasCopied) {
+			onCopyProp();
+		} else {
+			onFinishCopy();
+		}
+
+		lastHasCopied.current = hasCopied;
+	}, [hasCopied, onCopyProp, onFinishCopy]);
+
+	const handleOnClick = (event) => {
```

Comment:
@saramarcondes I've been memoizing callbacks G2 Components. I didn't originally... how important it was for [performance](https://github.com/ItsJonQ/g2/issues/57) 🙈 (assuming the consumer also passes in a memoized callback).

It's slightly tedious, but I think the trade off and value is worth it 🤞